stat.go

```diff
@@ -372,6 +372,16 @@ func HarmonicMean(x, weights []float64) float64 {
 	return math.Exp(math.Log(W) - v)
 }
 
+// CircularMean returns the circular mean of the dataset
+func CircularMean(x []float64) float64 {
```

Comment:
It'll just be `aX += w[i] * math.Cos(x[i])` if the weights aren't nil. Look at Mean for an example how to code.

Please move this function to be in alphabetical order in the file.